### PR TITLE
adds the SX-16 and removes the ZX

### PR DESCRIPTION
**Pull request by Spyroshark**
## About The Pull Request

Adds a new shotgun, the SX-16! It can be found in any squad vendor, and fires 16 gauge buckshot or slugs. It is significantly weaker then other shotguns, but makes up for it with a very fast fire rate. It uses magazines rather then being hand-loaded, and each magazine holds 15 shots. the 16 gauge variants of buckshot and slugs also don't have any stagger or stuns, for balance reasons.

Also removes the ZX from the game, but keeps it and the shotgun kit in the code in case someone wants to re-balance the ZX and re-add it later on.

sxshotgun
Thanks to Kuro for making the sprites!

## Why It's Good For The Game

The ZX is a mess to balance, and really just needs a removal at this point. Originally I was going to make this as a replacement for the shotgun spec but the concept of a "support shotgun" doesn't really work no matter how you balance it, and I think that making it a standard gun is just a better option.

NOTE: This will probably need some balance tweaks before it's ready for a full merge.

## Changelog
:cl:
add: Added the SX-16, a new marine shotgun! You can find it and it's ammo in any squad vendor.
imageadd: SX-16 sprites made by Kuro
del: Removed the ZX shotgun and the scout shotgun subkit.
/:cl:

### CODE REVIEW

**Comment by Rohesie** [comment]
Check the attachment offsets. The RDS at least looks a bit away from where it should be.

**Comment by Spyroshark** [comment]
looks like the RDS was off by a couple pixels, looks good now.

**Comment by spookydonut** [comment]
Do we need more ammo types?

**Comment by spookydonut** [comment]
spyro says they are still working on this

**Comment by Spyroshark** [comment]
alright i made some changes to buckshot and replaced slugs with flechette. This is ready to be tested again, hopefully it goes better this time.

**Comment by Spyroshark** [comment]
Made some balance changes and added slugs again, lets see how it is ingame now

File: code/modules/projectiles/updated_projectiles/ammo_datums.dm [hunk 1]
```diff
@@ -590,6 +590,64 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
...
+	hud_state = "shotgun_buckshot"
+	bonus_projectiles_type = /datum/ammo/bullet/shotgun/sx16_buckshot/spread
+	bonus_projectiles_amount = 2
+	bonus_projectiles_scatter = 10
+	accuracy_var_low = 10
+	accuracy_var_high = 10
+	max_range = 10
+	damage = 40
+	damage_falloff = 2
+	penetration = 10
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/ammo_datums.dm]: Does buckshot need pen?

File: code/modules/projectiles/updated_projectiles/ammo_datums.dm [hunk 2]
```diff
@@ -590,6 +590,64 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
...
+
+/datum/ammo/bullet/shotgun/sx16_buckshot/spread
+	name = "additional buckshot"
+	icon_state = "buckshot"
+	accuracy_var_low = 7
+	accuracy_var_high = 7
+	max_range = 10
+	damage = 30
+	damage_falloff = 2
+	penetration = 10
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/ammo_datums.dm]: Same for spread.

File: code/modules/projectiles/updated_projectiles/ammo_datums.dm [hunk 3]
```diff
@@ -590,6 +590,64 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
...
+
+/datum/ammo/bullet/shotgun/sx16_slug
+	name = "shotgun slug"
+	hud_state = "shotgun_slug"
+	shell_speed = 3
+	max_range = 15
+	damage = 40
+	penetration = 20
+
+/datum/ammo/bullet/shotgun/sx16_slug/on_hit_mob(mob/M,obj/item/projectile/P)
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/ammo_datums.dm]:
```suggestion
/datum/ammo/bullet/shotgun/sx16_slug/on_hit_mob(mob/M, obj/item/projectile/P)
```

File: code/modules/projectiles/updated_projectiles/ammo_datums.dm (hunk 1 above)
Comment (Spyroshark) [reply code/modules/projectiles/updated_projectiles/ammo_datums.dm]: i guess not, it barely has any so ill just remove it

**Comment by Spyroshark** [comment]
I think this PR is all fixed and finished now.

**Comment by Spyroshark** [comment]
are you kidding theres another merge conflict

**Comment by Spyroshark** [comment]
there THAT should fix the merge conflict

File: code/modules/projectiles/updated_projectiles/ammo_datums.dm [hunk 4]
```diff
@@ -590,6 +590,62 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
...
+/datum/ammo/bullet/shotgun/sx16_slug
+	name = "shotgun slug"
+	hud_state = "shotgun_slug"
+	shell_speed = 3
+	max_range = 15
+	damage = 40
+	penetration = 20
+
+/datum/ammo/bullet/shotgun/sx16_slug/on_hit_mob(mob/M, obj/item/projectile/P)
+	staggerstun(M, P, 5, 1, 0, 0, 0, 1)
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/ammo_datums.dm]:
```suggestion
	staggerstun(M, P, 5, 1, knockback = 1)
```

File: code/modules/projectiles/updated_projectiles/ammo_datums.dm [hunk 5]
```diff
@@ -590,6 +590,62 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
...
+/datum/ammo/bullet/shotgun/sx16_slug
+	name = "shotgun slug"
+	hud_state = "shotgun_slug"
+	shell_speed = 3
+	max_range = 15
+	damage = 40
+	penetration = 20
+
+/datum/ammo/bullet/shotgun/sx16_slug/on_hit_mob(mob/M, obj/item/projectile/P)
+	staggerstun(M, P, 5, 1, knockback = 1)
```
Comment (spookydonut) [inline code/modules/projectiles/updated_projectiles/ammo_datums.dm]:
```suggestion
	staggerstun(M, P, slowdown = 1, knockback = 1)
```

**Pull request merged**